### PR TITLE
indexer: split object version ingestion for perf

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

objects_history has been a bottleneck of indexer ingestion / backfill, splitting objects_version and let it run in parallel.
this might have been a factor of recent backfill perf regression.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs
Comment (lxfind) [inline crates/sui-indexer/src/handlers/checkpoint_handler.rs:273]: Any reason to do it here instead of inside the committer directly?

**Review by lxfind — COMMENTED** [review]
I assume this need to be rebased on the parking branch instead of main?

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs
Comment (gegaowp) [reply crates/sui-indexer/src/handlers/checkpoint_handler.rs:273]: I tend to group indexing codes, namely codes to get Stored* types in `index_checkpoint` while `committer` persists `Stored.*` into DB.

**Comment by gegaowp** [comment]
@lxfind

**Comment by lxfind** [comment]
> not really as this does not change any schema and only changes the binary itself, and thus can be deployed anytime, did I miss anything?

Well with this dev mode rebasing the parking branch to main will be very painful (and hence vulnerable to mistakes) down the road.

File: crates/sui-indexer/src/handlers/checkpoint_handler.rs
Comment (lxfind) [inline crates/sui-indexer/src/handlers/checkpoint_handler.rs:286]: Unnecessary clone?

**Review by lxfind — APPROVED** [review]

**Comment by gegaowp** [comment]
rebased the pr to breaking change park branch and will merge it when ci passes

**Pull request merged**
